crates/pallet-domains/src/staking.rs

```diff
@@ -187,10 +191,52 @@ fn freeze_account_balance_to_operator<T: Config>(
     Ok(())
 }
 
+pub(crate) fn do_switch_operator_domain<T: Config>(
+    operator_owner: T::AccountId,
+    operator_id: OperatorId,
+    new_domain_id: DomainId,
+) -> Result<DomainId, Error> {
+    ensure!(
+        OperatorIdOwner::<T>::get(operator_id) == Some(operator_owner),
+        Error::NotOperatorOwner
+    );
+
+    ensure!(
+        DomainStakingSummary::<T>::contains_key(new_domain_id),
+        Error::DomainNotInitialized
+    );
+
+    OperatorPools::<T>::try_mutate(operator_id, |maybe_operator_pool| {
+        let operator_pool = maybe_operator_pool.as_mut().ok_or(Error::UnknownOperator)?;
+        operator_pool.next_domain_id = new_domain_id;
+
+        // remove operator from next_operators from current domains.
+        // operator is added to the next_operators of the new domain once the
+        // current domain epoch is finished.
+        DomainStakingSummary::<T>::try_mutate(
+            operator_pool.current_domain_id,
```

Comment:
Seems this check is not added.

Reply:
This is added here - [`4a221da` (#1646)](https://github.com/subspace/subspace/pull/1646/commits/4a221dab6327b6e32d80803a0021467da3aea816#diff-bc31bd26a15ac3e6de45a5242e5fc7883136b699fa411a472d0d3eb45c50f0f9R213)

Not sure what you meant otherwise

Reply:
`operator.current_domain_id == operator.next_domain_id` check pending switch but I mean `operator.current_domain_id != new_domain_id `.

Reply:
Ahh seems like I forgot to add that. Instead of error, for consistency, I'll just make it noop. Thanks!